helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -84,6 +88,9 @@ public AccessResult() {
   private static Logger LOG = LoggerFactory.getLogger(ZkBaseDataAccessor.class);
 
   private final HelixZkClient _zkClient;
+  // onDemand zkClient for read/write non-znRecord data path;
+  // minimal change to support custom serializer
```

Comment:
This comment is confusing and not necessary. Let's remove this. Or just say

"ZkClient that will be lazily instantiated if there are operations for non-ZNRecord data"

Reply:
updated.